src/components/RewardsWrapper/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { Fragment } from 'react'
+import { useGetQuest } from '@/hooks/useGetQuest'
+import { useGetRewards } from '@/hooks/useGetRewards'
+import { useGetUserPlayStreak } from '@/hooks/useGetUserPlayStreak'
+import { useQuestWrapper } from '@/state/QuestWrapperProvider'
+import { RewardsRow, Rewards } from '@hyperplay/ui'
+import { RewardWrapper } from '../RewardWrapper'
+import { UseGetRewardsData } from '@/types/quests'
+
+export function RewardsWrapper({ questId }: { questId: number | null }) {
+  const { getQuest, logError, getUserPlayStreak, getExternalTaskCredits } =
+    useQuestWrapper()
+
+  const { data: questQuery } = useGetQuest(questId, getQuest)
+
+  const {
+    data: questPlayStreakQuery,
+    invalidateQuery: invalidateQuestPlayStreakQuery
+  } = useGetUserPlayStreak(questId, getUserPlayStreak)
+
+  const { data: rewardsQuery } = useGetRewards({
+    questId,
+    getQuest,
+    getExternalTaskCredits,
+    logError
+  })
+
+  const questMeta = questQuery?.data
+  const questPlayStreakData = questPlayStreakQuery?.data
+  const rewardsData = rewardsQuery?.data
+
+  if (!rewardsData || !questMeta) {
+    return null
+  }
+
+  // Organize rewards by category
+  const rewardsByCategory: Record<string, UseGetRewardsData[]> = {}
+
+  for (const reward_i of rewardsData) {
+    if (Object.hasOwn(rewardsByCategory, reward_i.chainName)) {
+      rewardsByCategory[reward_i.chainName].push(reward_i)
+    } else {
+      rewardsByCategory[reward_i.chainName] = [reward_i]
+    }
+  }
```

Comment:
wondering if we should make a helper for this and use it/return `rewardsByCategory` from `useGetRewards`. It's O(n) so probably fine to just run it every time `useGetRewards` runs even if the consuming code doesn't necessarily need the map.